src/app/admin/mod.rs

```diff
@@ -55,22 +69,110 @@ where
     type ReqBody = Body;
     type ResBody = Body;
     type Error = io::Error;
-    type Future = FutureResult<Response<Body>, Self::Error>;
+    type Future = ResponseFuture;
 
     fn call(&mut self, req: Request<Body>) -> Self::Future {
+        let span = trace_span!(
+            "admin",
+            request.path = %req.uri().path(),
+            request.method = ?req.method(),
+        );
+        let _enter = span.enter();
```

Comment:
Would you mind splitting out the part of this change that adds spans to the admin endpoint to a followup?